src/create-reminder-signup/lib/fetch-with-retry.test.ts

```diff
@@ -0,0 +1,39 @@
+import { Response } from 'node-fetch';
+import { fetchWithRetry } from './fetch-with-retry';
+
+describe('fetchWithRetry', () => {
+	it('retries the fetcher if it fails', async () => {
+		const numFails = 1;
+		const maxRetries = 5;
+		const expectedNumTries = 2;
+		const fetcher = getMockFetcher(numFails);
+
+		await fetchWithRetry(fetcher, maxRetries);
+
+		expect(fetcher).toHaveBeenCalledTimes(expectedNumTries);
+	});
+
+	it('only retries maxRetries times', async () => {
+		const numFails = 5;
+		const maxRetries = 1;
+		const expectedNumTries = 2;
+		const fetcher = getMockFetcher(numFails);
+
+		await fetchWithRetry(fetcher, maxRetries);
+
+		expect(fetcher).toHaveBeenCalledTimes(expectedNumTries);
+	});
+});
```

Comment:
These might not be the most crucial tests ever, but I was quite happy to get a test in there to stop e.g an off by one error

Reply:
🥇